src/main/java/tutoraid/model/StudentBook.java

```diff
@@ -94,11 +107,49 @@ public void removeStudent(Student key) {
         students.remove(key);
     }
 
+    //// lesson-level operations
+
+    /**
+     * Returns true if a lesson with the same identity as {@code lesson} exists in the address book.
+     */
+    public boolean hasLesson(Lesson lesson) {
```

Comment:
I think we may need a LessonBook to handle all the methods related to Lesson.

Reply:
My suggestion* is to have a Book superclass, then StudentBook and LessonBook extending from it, in case you can't separate them (*I am not very familiar with the implementation of this so my suggestion might be unviable)